pkg/codegen/schema.go

```diff
@@ -52,6 +62,52 @@ func (s Schema) GetAdditionalTypeDefs() []TypeDefinition {
 	return result
 }
 
+func (s Schema) EnumValues() []Enum {
+	if s.GoType != "string" && !strings.HasPrefix(s.GoType, "int") {
+		return nil
+	}
+	if s.Extensions == nil {
+		return nil
+	}
+	if _, ok := s.Extensions["enum"]; !ok {
```

Comment:
all extensions must be of the form `x-something`, or openapi validation of the spec will fail.